secret-engine/src/main/java/com/sequenceiq/cloudbreak/service/secret/service/rotation/context/VaultRotationContext.java

```diff
@@ -1,21 +1,21 @@
 package com.sequenceiq.cloudbreak.service.secret.service.rotation.context;
 
 import java.util.Map;
+import java.util.function.Supplier;
 
 import com.sequenceiq.cloudbreak.rotation.secret.RotationContext;
-import com.sequenceiq.cloudbreak.service.secret.domain.Secret;
 
 public class VaultRotationContext extends RotationContext {
 
-    private final Map<Secret, String> newSecretMap;
+    private final Map<String, Supplier<String>> secretUpdateSupplierMap;
```

Comment:
Instead of `Map<String, Supplier<String>>` it could be a `Map<String, Class<? extends SecretGenerator>>`. This has the advantage that we can serialize/deserialize it to/from json. SecretGenerator can be an interface like:

```java
public interface SecretGenerator {

  String generate();
}
```